### DISCUSSION TITLE
Which sensors do we consider 'Diagnostic'?

### DISCUSSION THREAD

**Discussion by benjamin-dcs** [jasperslits/haithowifi#64] [General]
Copy-Paste this list and check diagnostic sensors
```markdown
### Autotemp
- [ ] Empty battery
- [ ] Error
- [ ] Mode
- [ ] Room X power % (%)
- [ ] Room X power kW (kW)
- [ ] Room X setpoint
- [ ] Room X temp
- [ ] Status \| Cod
### CVE
- [ ] Error
- [ ] Fan Setpoint (rpm)
- [ ] Fan Speed (rpm)
- [ ] Filter dirty
- [ ] Humidity
- [ ] Temperature
- [ ] Total Operating Time
- [ ] Ventilation setpoint (%)
- [ ] Last Command
- [ ] Last Command Source
### HRU
- [ ] Actual Mode
- [ ] Air Quality (%)
- [ ] Airfilter counter
- [ ] Balance (%)
- [ ] Bypass position
- [ ] Exhaust fan (RPM)
- [ ] Exhaust temp (°C)
- [ ] Global fault code
- [ ] Highest received CO2 value (Ppm) 
- [ ] Highest received RH value (%RH)
- [ ] Remaining override timer (Sec)
- [ ] Supply fan (RPM)
- [ ] Supply temp (°C)
- [ ] Last Command
- [ ] Last Command Source
### WPU
- [ ] Boiler pump (%)
- [ ] Boiler temp up (°C)
- [ ] CV pressure (Bar)
- [ ] Cv pump (%)
- [ ] CV return temp (°C)
- [ ] Flow sensor (lt_hr)
- [ ] Heat demand thermost. (%)
- [ ] Status
- [ ] Temp from source (°C)
- [ ] Temp to source (°C)
- [ ] Requested room temp (°C)
- [ ] Room temp (°C)
- [ ] Well pump (%)
```

**Comment by jasperslits** [comment]
Not exact science 

### Autotemp
- [X] Empty battery
- [X] Error
- [ ] Mode
- [ ] Room X power % (%)
- [ ] Room X power kW (kW)
- [ ] Room X setpoint
- [ ] Room X temp
- [ ] Status \| Cod
### CVE
- [X] Error
- [ ] Fan Setpoint (rpm)
- [ ] Fan Speed (rpm)
- [X] Filter dirty
- [ ] Humidity
- [ ] Temperature
- [X] Total Operating Time
- [ ] Ventilation setpoint (%)
- [X] Last Command
- [X] Last Command Source
### HRU
- [ ] Actual Mode
- [ ] Air Quality (%)
- [X] Airfilter counter
- [ ] Balance (%)
- [ ] Bypass position
- [ ] Exhaust fan (RPM)
- [ ] Exhaust temp (°C)
- [X] Global fault code
- [ ] Highest received CO2 value (Ppm) 
- [ ] Highest received RH value (%RH)
- [X] Remaining override timer (Sec)
- [ ] Supply fan (RPM)
- [ ] Supply temp (°C)
- [X] Last Command
- [X] Last Command Source
### WPU
- [ ] Boiler pump (%)
- [ ] Boiler temp up (°C)
- [ ] CV pressure (Bar)
- [ ] Cv pump (%)
- [ ] CV return temp (°C)
- [X] Flow sensor (lt_hr)
- [ ] Heat demand thermost. (%)
- [ ] Status
- [ ] Temp from source (°C)
- [ ] Temp to source (°C)
- [ ] Requested room temp (°C)
- [ ] Room temp (°C)
- [ ] Well pump (%)

**Comment by benjamin-dcs** [comment]
Here are distinct and non-overlapping definitions for the three categories of entities in Home Assistant:

### 1. **Sensor**  
   **Definition:**  
   A **sensor** entity represents **passive, read-only data** provided by a device. It reflects the current state or measurement of something in the environment without directly interacting with or changing the device.  

   **Examples:**  
   - Temperature readings (e.g., `sensor.living_room_temperature`)  
   - Energy consumption data (e.g., `sensor.total_energy_usage`)  
   - Weather conditions (e.g., `sensor.outdoor_humidity`)  
   - Motion detection state (e.g., `sensor.garage_motion`)  

   **Key Characteristics:**  
   - Data is updated by the device.  
   - Does not accept user input or commands.  
   - Represents environmental or device status metrics.

---

### 2. **Control**  
   **Definition:**  
   A **control** entity represents **interactive elements** that allow the user to send commands or configure a device to change its state or behavior.  

   **Examples:**  
   - Light switches (e.g., `light.kitchen_light`)  
   - Thermostat controls (e.g., `climate.bedroom_thermostat`)  
   - Media player controls (e.g., `media_player.living_room_tv`)  
   - Lock mechanisms (e.g., `lock.front_door`)  

   **Key Characteristics:**  
   - Enables direct interaction with the device.  
   - Changes the state or operation of the device when a command is sent.  
   - May involve toggles, sliders, or discrete command actions.

---

### 3. **Diagnostic**  
   **Definition:**  
   A **diagnostic** entity provides **device-specific metadata or operational insights** that assist in understanding the device's internal state or functioning but is not directly related to the user's environment.  

   **Examples:**  
   - Battery level (e.g., `sensor.battery_level`)  
   - Signal strength (e.g., `sensor.wifi_signal_strength`)  
   - Error or status codes (e.g., `sensor.device_error_state`)  
   - Device uptime (e.g., `sensor.device_uptime`)  

   **Key Characteristics:**  
   - Focused on the health or internal operation of the device.  
   - Typically supports troubleshooting or maintenance.  
   - Not directly actionable by the user.  

By defining entities this way, **sensors** describe the world, **controls** allow changes to the world, and **diagnostics** describe the device itself.

**Comment by benjamin-dcs** [comment]
I was doubting about ’filter_dirty’, guess that one can go everywhere. Let's go with your list 👍
